src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -372,6 +376,8 @@ private AIState lookForNeededItems()
 
     /**
      * Updates the itemsCurrentlyNeeded with current values.
+     * 
+     * This function seems bugged, as it gives itemsCurrentlyNeeded = isEmpty 
```

Comment:
The code explicitly states that if `itemsCurrentlyNeeded ` is empty then `job.getItemsNeeded()` is also empty and maybe thats the problem. 

We first add all `itemsNeeded` items to the job, then remove all items from the workers inventory because he already has them. And then copy that to his current needs. 

So you may need to check if `itemsNeeded` is empty. Because then this method does what it should. If this is empty then there are no needed items. If it is not empty then there is something really weird going on...